tests/unit/test_collector.py

```diff
@@ -189,10 +191,43 @@ def test_get_machine_type(self, collector_daemon, mac_address, expect_machine_ty
                 }
             }
         }
-        machine_type = statsd._get_machine_type(machine)
+        machine_type = statsd._get_machine_type(machine, "dummy-0")
 
         assert machine_type.value == expect_machine_type
 
+    def test_get_machine_type_interface_skip(self, collector_daemon):
+        """Test that blacklisted interfaces are not used to detect machine type."""
+        statsd = collector_daemon()
+        skip_interfaces = ["virb*", "tap-*"]
+        kvm_prefix = "fa:16:3e:"
+        tap_prefix = "52:54:00:"
+        machine_mac = "00:00:00:00:00:00"
+
+        machine = {
+            "network-interfaces": {
+                "ens3": {
+                    "mac-address": machine_mac,
+                },
+                "virbr0": {
+                    "mac-address": kvm_prefix + "00:00:01",
+                },
+                "virbr1": {
+                    "mac-address": kvm_prefix + "00:00:02",
+                },
+                "tap-0": {
+                    "mac-address": tap_prefix + "00:00:01",
+                },
+            }
+        }
+
+        statsd.config["detection"]["virt_macs"].set([kvm_prefix, tap_prefix])
+        # Assert that without skipping interfaces, machine is detected as virtual
+        assert statsd._get_machine_type(machine, "dummy-0") == MachineType.KVM
```

Comment:
it's better to have a single assert per test (so it's clearer what broke when one test fails). I would suggest splitting this test into two (perhaps parametrizing if convenient)